mesonbuild/modules/python.py

```diff
@@ -297,6 +300,55 @@ def __init__(self, interpreter: 'Interpreter') -> None:
             'find_installation': self.find_installation,
         })
 
+    def _get_install_scripts(self) -> T.List[mesonlib.ExecutableSerialisation]:
+        backend = self.interpreter.backend
+        ret = []
+        optlevel = self.interpreter.environment.coredata.get_option(mesonlib.OptionKey('bytecompile', module='python'))
+        if optlevel == -1:
+            return ret
+
+        installdata = backend.create_install_data()
+        py_files = []
+
+        def should_append(f, isdir: bool = False):
+            return f.startswith(('{py_platlib}', '{py_purelib}')) and (f.endswith('.py') or isdir)
+
+        for t in installdata.targets:
+            if should_append(t.out_name):
+                py_files.append((t.out_name, os.path.join(installdata.prefix, t.outdir, os.path.basename(t.fname))))
+        for d in installdata.data:
+            if should_append(d.install_path_name):
+                py_files.append((d.install_path_name, os.path.join(installdata.prefix, d.install_path)))
+        for d in installdata.install_subdirs:
+            if should_append(d.install_path_name, True):
+                py_files.append((d.install_path_name, os.path.join(installdata.prefix, d.install_path, '')))
```

Comment:
Why `''` at the end?

Reply:
This forces directories to be emitted with a trailing directory separator.

Reply:
Why is that needed? Please add a comment there because it's not obvious.

Reply:
It is not needed anymore because the bytecompile script just checks `os.path.isdir()` now. ;) Removing.